src/metabase/task/QUARTZ.md

```diff
@@ -0,0 +1,153 @@
+# Quartz Scheduler in Metabase
+
+This document provides guidance for developers on how to use the Quartz scheduler in Metabase for background task scheduling.
+
+## Overview
+
+Metabase uses [Quartz](http://www.quartz-scheduler.org/) for scheduling and executing background tasks. The Quartz scheduler is initialized during Metabase startup and manages various tasks like syncing databases, sending pulses, cleaning up sessions, and more.
+
+In Quartz, there are two fundamental concepts:
+
+- **Job**: Represents a unit of work to be executed. Jobs are defined as classes that implement specific interfaces and contain the actual code to be run.
+- **Trigger**: Defines when a job should be executed - it can be based on a schedule (like a cron expression), a specific time, or in response to events.
+
+Jobs and triggers are created separately and then associated with each other when scheduled in the Quartz system.
+
+## Use cases at Metabase
+
+This list is not complete and is open to additions. If you have other use cases or patterns for Quartz tasks in Metabase, please feel free to add them here.
+
+### 1. One-Time Startup Tasks
+
+These tasks execute once when Metabase starts up. They're useful for initialization operations.
+
+```clojure
+(ns my.startup.task
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [clojurewerkz.quartzite.schedule.simple :as simple]
+   [metabase.task :as task]))
+
+(def startup-job-key     (jobs/key "my.startup.task.job"))
+(def startup-trigger-key (triggers/key "my.startup.task.trigger"))
+
+(jobs/defjob StartupTask [_]
+  (println "Running startup initialization task"))
+
+(defmethod task/init! ::StartupTask [_]
+  (let [job (jobs/build
+             (jobs/of-type StartupTask)
+             (jobs/with-identity startup)
+             (jobs/with-description "One-time startup initialization task")
+             (jobs/store-durably))
+        trigger (triggers/build
+                 (triggers/with-identity startup)
+                 (triggers/for-job startup)
+                 (triggers/start-now))]
+    (task/schedule-task! job trigger)))
```

Comment:
It would be great to have a comment on each line explaining the significance of each one.

Also, why do we need this job to be durable?

Reply:
added the comments.

About `store-durably` yea I don't think we need it since we'll re add the job on every startup. Can remove.